libs/SalesforceSDK/src/com/salesforce/androidsdk/security/SalesforceKeyGenerator.java

```diff
@@ -94,7 +97,12 @@ public static String getUniqueId(String name, int length) {
      * @return Encryption key.
      */
     public static String getEncryptionKey(String name) {
-        return generateEncryptionKey(name);
+        String encryptionKey = CACHED_ENCRYPTION_KEYS.get(name);
```

Comment:
Bringing back the code we had in 7.0 and before

Reply:
Thanks for bringing this back. Should the map be a ConcurrentHashMap? I guess this was working pre 7.1, so it is ok.